src/shared/utils/file-operations.js

```diff
@@ -95,23 +91,29 @@ export function saveFileToServer(
   fileID = undefined
 ) {
   const formData = makeFormData(notebookID, data, fileName);
-  // FIXME: we should not have update & load be separate files,
-  // from an API standpoint. We should just have one file upload function
-  // that deals with the particulars.
-  const fetchRequest = fileID // if fileID undefined, upload, otherwise update
-    ? fd => updateFile(fileID, fd)
-    : fd => uploadFile(fd);
-  return fetchRequest(formData).then(r => {
+  return uploadFile(formData, fileID).then(r => {
     if (r.status === 500) throw new Error(r.statusText);
     return r.json();
   });
 }
 
-export function deleteFileOnServer(fileID) {
-  return fetchWithCSRFTokenAndJSONContent(`/api/v1/files/${fileID}/`, {
-    method: "DELETE"
-  }).then(r => {
+export function makeDeleteRequest(url) {
```

Comment:
most delete requests have the exact same structure, with the difference being the URL in which we suggest to delete. This function should not ever be used – rather, I would suggest following the approach used in eg `deleteNotebookOnServer` below.

Reply:
if it should not be used, should we not export it? or if we need to export it for testing, should we call it something like `makeDeleteRequest_private` and adopt a convention around not using thusly named vars outside of the file that defines them?